src/components/tabController/TabPage.tsx

```diff
@@ -47,13 +49,25 @@ export default function TabPage({
   }, [lazy, shouldLoad]);
 
   useAnimatedReaction(() => {
-    return targetPage.value === index;
+    return currentPage.value;
   },
-  isActive => {
+  (currentPage, previousPage) => {
+    const isActive = currentPage === index;
+    const wasActive = previousPage === index;
+    const nearActive = asCarousel && (currentPage - 1 === index || currentPage + 1 === index);
+    const wasNearActive = asCarousel && previousPage && (previousPage - 1 === index || previousPage + 1 === index);
```

Comment:
I think `previousPage` should be `previousPage !== null` (for when it's `0`).

Reply:
Right! 
Fixed